### PR TITLE
Cart Checkout Confirmation Screen

### PR DESCRIPTION
**Pull request by ashi72**
besides the confirmation screen, also added two buttons for navigating to the checkout screen and changed the stack navigator; also installed some new packages like one for icons (react native vector icons);

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { AppRegistry } from 'react-native';
+import Config from 'react-native-config';
 import App from './src/App';
 import { name as appName } from './app.json';
 
+console.log(Config.REACT_APP_AIRTABLE_BASE_KEY);
```
Comment (curesh) [inline index.js:6]: Can we remove this line, and also the import above

File: src/Checkout/CheckoutScreen.js [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    backgroundColor: '#0492c2',
+  },
+  topIcon: {
+    height: 70,
+    marginHorizontal: 8.0,
+    paddingTop: 12.0,
+  },
+  title: {
+    fontSize: 18,
+    // marginLeft: '2%',
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:35]: Remove unneeded comments here and in the rest of the file

File: src/Checkout/CheckoutScreen.js [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    marginLeft: 5,
+    marginRight: 5,
+    fontSize: 20,
+  },
+  button: {
+    width: '30%',
+    marginTop: 10,
+    backgroundColor: '#0492c2',
+  },
+  topIcon: {
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:28]: Remove since its not used

File: src/Profile/ProfileScreen.js [hunk 4]
```diff
@@ -22,18 +24,77 @@ const styles = StyleSheet.create({
...
+  .base(airtableConfig.baseKey);
+
+// const temporaryUser = { name: 'Jordan', email: 'gmail@gmail.com', address: '1200 covid begone' };
+
 export default function ProfileScreen({ navigation }) {
+  const [profile, setProfile] = useState([]);
+
+  const getUserInfo = () => {
+    base('Users').select({ maxRecords: 1, filterByFormula: "({email}='aaronqqshi@gmail.com')" }).firstPage()
+    // base('Users').select({
```
Comment (curesh) [inline src/Profile/ProfileScreen.js:49]: Remove comments that you dont need

File: src/Profile/ProfileScreen.js [hunk 5]
```diff
@@ -22,18 +24,77 @@ const styles = StyleSheet.create({
...
+    //   filterByFormula: '{email} = "aaronqqshi@gmail.com"',
+    // })
+    // base('Users').select({ view: 'Grid view' }).all()
+      .then((records) => {
+        console.log(records);
+        setProfile(records[0].fields);
+      });
```
Comment (curesh) [inline src/Profile/ProfileScreen.js:56]: Add error handler for airtable call, and do something like "Alert.alert(err.message);" inside of it

File: src/Checkout/CheckoutScreen.js [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+export default function CheckoutScreen({ navigation }) {
+  const [shippingAddress, setShippingAddress] = useState([]);
+  const [total, setTotal] = useState(0);
+  const [count, setCount] = useState(0);
+  const [deliveryDate, setDeliveryDate] = useState('unavailable');
+  const [deliveryFee, setDeliveryFee] = useState(0);
+  const [freeFee, setFreeFee] = useState(0);
+
+  const calcTotal = () => {
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:69]: Modify this so it takes in a parameter user email variable, and plugs that into the airtable call

File: src/Checkout/CheckoutScreen.js [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+          c += 1;
+        });
+        setTotal(sum);
+        setCount(c);
+      });
+  };
+
+  useEffect(() => {
+    base('Users UPDATED').select({
+      filterByFormula: '{email} = "helen@gmail.com"',
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:85]: modify this to use the same email variable as above, and define this at the top of the function

File: src/Checkout/CheckoutScreen.js [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+            address: records[0].fields.address,
+            zipcode: records[0].fields['zip code'],
+            apartmentLine: '',
+          });
+        }
+      });
+    calcTotal();
+
+    base('Orders').select({
+      filterByFormula: '{user_id} = "helen@gmail.com"',
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:105]: same thing as above

File: src/Checkout/CheckoutScreen.js [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+        if (records[0].fields['delivery date (temp)'] !== '') {
+          setDeliveryDate(records[0].fields['delivery date (temp)']);
+        }
+      });
+  }, []);
+
+  return (
+    <View>
+      <View style={[styles.subcontainer, {
+        marginTop: '8%', marginHorizontal: '8%', marginBottom: '4%', borderBottomColor: 'grey', borderBottomWidth: 2,
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:121]: Move the styling attributes into the subcontainer thing in the stylesheet

File: src/Checkout/CheckoutScreen.js [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+      }]}
+      >
+        <Text style={[styles.title, { fontWeight: '700', marginLeft: '0%', marginBottom: '0%' }]}>
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:124]: same as above

File: src/Checkout/CheckoutScreen.js [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+          mode="contained"
+          style={styles.button}
+          onPress={() => navigation.navigate('Marketplace')}
+        >
+          MARKET
+        </Button>
+        <Button
+          mode="contained"
+          style={styles.button}
+          onPress={() => navigation.navigate('Marketplace')}
```
Comment (curesh) [inline src/Checkout/CheckoutScreen.js:241]: Add a place order button instead of the buy tbd

**Review by curesh — CHANGES_REQUESTED** [review]
Wow this feature is so clean, the styling looks almost exactly like the figma! Your code is also super clean as always. I've added a few comments about general syntactical stuff, and removing unneeded comments. Also could you add the place order button at the bottom, and remove the market button (since it does the same thing). Otherwise, great work and I'm looking forward to seeing what else you're gonna do this year!

File: src/Checkout/CheckoutScreen.js (hunk 9 above)
Comment (ashi72) [reply src/Checkout/CheckoutScreen.js:121]: even if its just used once?

**Comment by ashi72** [comment]
mostly done! not sure about the subcontainer styling header if its a one time specification and theres other subcontainers with different specifications? also went back and fixed issues from the profile screen pr hehe

**Pull request closed without merging**